glad/generator/rust/templates/impl.rs

```diff
@@ -12,6 +12,16 @@ pub struct FnPtr {
 }
 
 impl FnPtr {
+    {% if options.mx %}
+    pub fn new<F>(loadfn: &mut F, name: &'static str) -> FnPtr where F: FnMut(&'static str) -> *const raw::c_void {
```

Comment:
What about passing the `raw::c_void` as argument instead of the loader and the name (the loading logic shouldn't be associated with the `FnPtr` imo).

```rust
FnPtr::new(load("glClear"))
```